mne/datasets/utils.py

```diff
@@ -226,196 +157,93 @@ def _do_path_update(path, update_path, key, name):
+        for pooch.
+    check_version : bool
+        Whether to check the version of the dataset or not. Each version
+        of the dataset is stored in the root with a ``version.txt`` file.
+    return_version : bool
+        Whether or not to return the version of the dataset or not.
+        Defaults to False.
+    accept : bool
+        Some datasets require an acceptance of an additional license.
+        Default to False. If this is True, then license text should
+        be passed into key word argument ``license_text``.
+    license_text : str | None
+        The text of a license agreement. Only used if ``accept`` is True.
+
+    Returns
+    -------
+    path : str
+        Path to {name} dataset directory.
+    """
+    # get download path for specific dataset
+    path = _get_path(path=path, key=CONFIG_KEYS[name], name=name)
+
+    # get the actual path to each dataset folder name
+    final_path = op.join(path, FOLDER_NAMES[name])
+
+    # handle BrainStorm datasets with nested folders for datasets
+    if name.startswith('bst_'):
+        final_path = op.join(final_path, name)
+
     # additional condition: check for version.txt and parse it
-    want_version = releases.get(name, None)
+    # check if testing or misc data is outdated; if so, redownload it
+    want_version = RELEASES.get(name, None)
     want_version = _FAKE_VERSION if name == 'fake' else want_version
-    if not need_download and want_version is not None:
-        data_version = _dataset_version(folder_path[0], name)
-        need_download = LooseVersion(data_version) < LooseVersion(want_version)
-        if need_download:
-            logger.info(f'Dataset {name} version {data_version} out of date, '
-                        f'latest version is {want_version}')
-    if need_download and not download:
-        return ''
-
-    if need_download or force_update:
-        logger.debug('Downloading: need_download=%s, force_update=%s'
-                     % (need_download, force_update))
-        for f in folder_path:
-            logger.debug('  Exists: %s: %s' % (f, op.exists(f)))
-        if name == 'brainstorm':
+
+    # get the version of the dataset and then check if the version is outdated
+    data_version = _dataset_version(final_path, name)
+    outdated_dataset = want_version is not None and \
+        LooseVersion(want_version) > LooseVersion(data_version)
+
+    if outdated_dataset:
+        logger.info(f'Dataset {name} version {data_version} out of date, '
+                    f'latest version is {want_version}')
+
+    # reasons to bail early (hf_sef has separate code for this):
+    if (not force_update) and (not outdated_dataset) and \
+            (not name.startswith('hf_sef_')):
+        # if target folder exists (otherwise pooch downloads every time,
+        # because we don't save the archive files after unpacking)
+        if op.isdir(final_path):
+            return final_path
```

Comment:
@adam2392 this is why the RuntimeWarning is not getting raised the second time we call `datasets._fake.data_path()` in the test. the `_do_path_update` is never getting reached because we return here.

Reply:
Thanks for pointing this out. I think I fixed it now.